tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -323,6 +323,19 @@ class Restrictions {
       ];
     }
 
+    var duplicatedField = _findFieldWithDuplicatedName(currentModel);
+    var ancestorWithDuplicatedField =
+        _findAncestorWithDuplicatedFieldName(currentModel);
+
+    if (duplicatedField != null && ancestorWithDuplicatedField != null) {
+      return [
+        SourceSpanSeverityException(
+          'You cannot declare "${currentModel?.className}" with field: "${duplicatedField.name}" as it is already declared in "${ancestorWithDuplicatedField.className}".',
```

Comment:
Suggestion: Make this validation error show on the field that is the duplicate and for each fields that is duplicate.

If I understand the change here correctly this error will show on the "extends" keyword.
It should instead be displayed on the field that is duplicate.

This achieves two things:
1. Allows us to guide the developer to the right spot around what needs to change.
2. Handle if multiple fields are duplicates in a nice way.